### ISSUE TITLE
portfolio access control should be a configuration property

### ISSUE THREAD

**Issue by mischa-n** [DependencyTrack/dependency-track#2609]
### Current Behavior

The "portfolio access control" is switchable at runtime (using the UI or the REST-API).

### Proposed Behavior

Enabling or disabling the "portfolio access control" should be an initial decision that is only configurable in the API server backend.

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this enhancement was already requested

**Comment by syalioune** [comment]
> Enabling or disabling the "portfolio access control" should be an initial decision that is only configurable in the API server backend

I disagree with that statement.  As long as the switch is protected behind administrative privilege, I don't think this is more an issue than an admin being able to change it in the API server backend. You can even remove `ACCESS_MANAGEMENT` from administrators to prevent unwanted toggle if need be (and add it back using SQL requests).

**Comment by mischa-n** [comment]
Would be nice to get a second opinion.

I thought that the goal of this application is to provide a platform for projects with separate teams and project specific roles.
Other related products like SonarQube, Bamboo, Github, Gitlab, ... do not have such a switch that may lead to a situation where a permission in a project may become a global permission.

Another thought: your permission UI is too laborious. It takes too long for a project admin to grant permissions to a new project member.
(My preferred) solutions from other products:

- Gitlab with its (fixed) role based permission model, may handle project hierarchies and has time limited grants
- SonarQube with explicite permissions but the fastest and easiest permission UI

**Comment by msymons** [comment]
@mischa-n, the work on portfolio access control is not yet regarded as complete...  there are 2 open issues that are currently assigned to the v4.9 milestone.

Suggestions for further improvement are always welcome.  Even more welcome are PRs!

Bear in mind that Dependency-Track is an OSS project.  All of the products that you mention (SonarQube, Bamboo, Github, Gitlab) are commercial...  which tends to have an impact on "what can be done and how quickly".  It is always useful to look at such products for inspiration about how they approach dealing with certain challenges...  but doing something similar (better) in DT is dependent on whether someone is prepared to step up and put in the time/effort.

That also means that implementation can be very quick indeed (faster than you might see in a commercial product) if you should submit a PR yourself or compose an enhancement that is worded in a way that gets someone else interested in doing the work.

**Comment by mischa-n** [comment]
We are working on getting more involved in OOS projects. As a first step we want to give more feedback, so here we are.
From a company's perspective, the issue of how to contribute is not as simple as it seems. It is often easier to buy a product (and cheaper).
But we understand that we definitely need to do more here.

Most people who make suggestions here (on GitHub) will do so as part of their job. These are either suggestions for improvement - if the respective software is already in use, or problems that prevent its use in their company.
At the same time, these people are not always the best candidates to contribute solutions themselves in the form of PRs.

We are not currently using DependencyTrack, but the dependency-check-maven plugin. We have exactly the problems with the plugin that you want to solve with your application.
When we deploy DependencyTrack at our site, we will have <10 admins and ~100 projects, each with 5..30 sub-applications and ~2 maintainers (at the project level).
So, from our perspective, rights management needs to reside with the maintainers and be so simple and clear that little to no support is required.
At the moment, this is our main obstacle to using DependencyTrack.

PS: SonarQube ist still Open Source...
